server/core_group.go

```diff
@@ -996,12 +996,16 @@ func ListGroups(ctx context.Context, logger *zap.Logger, db *sql.DB, name string
 	}
 
 	var query string
-	params := []interface{}{limit}
+	params := []interface{}{limit, omit_full}
+
 	if name == "" {
 		query = `
 SELECT id, creator_id, name, description, avatar_url, state, edge_count, lang_tag, max_count, metadata, create_time, update_time
 FROM groups
-WHERE disable_time = '1970-01-01 00:00:00 UTC'
+WHERE
+	(disable_time = '1970-01-01 00:00:00 UTC')
+AND
+	((NOT $2) OR (edge_count < max_count))
```

Comment:
I think this addition (`OR (edge_count < max_count)`) to the query will have a significant impact on the performance profile of the query. 

I ask, because the original query was a partial k-v look up and was hitting indexed fields as much as possible. Additionally the use of `OR` and range queries does not allow the database to fast-skip records on disk.

Have you performed `ANALYSIS ... ` on this query?

Reply:
Yes, I've seen the query analysis, and of course its performance may only be worse than it used to be without this additional filter. Although it doesn't seem more than marginally so when I try running these queries on a database with 10k entries in the `groups` table. Would that be in any way indicative of the real use conditions?

Anyway, it'll only matter for the queries with `omit_full=true` and after all the other filters on indexed fields have been applied. Unfortunately, I do not know much about the typical loads, so it's hard to judge whether an additional index is worth it for this kind of a query.

I've also studied the CockroachDB documentation, and it doesn't seem to say anything about issues with skipping ranges when using this or that SQL feature. Could you please explain what you meant in the second paragraph?